### ISSUE TITLE
Windows: os.stat() returns random times on ERROR_ACCESS_DENIED

### ISSUE THREAD

**Issue by danny0838** [python/cpython#114273]
# Bug report

### Bug description:

When retrieving the timestamp of an inaccessible file (or directory) in Windows, such as:

```python
os.path.getmtime('C:\\System Volume Information')
```

, a negative value such as `-11644246396.230042` is returned in Python 3.12.

In Python < 3.12, a normal timestamp such as `1705600376.5800736`  is returned instead.

This may break many scripts as most of them don't expect a negative value.

### CPython versions tested on:

3.12

### Operating systems tested on:

Windows

**Comment by serhiy-storchaka** [comment]
```
>>> time.gmtime(1705600376.5800736)
time.struct_time(tm_year=2024, tm_mon=1, tm_mday=18, tm_hour=17, tm_min=52, tm_sec=56, tm_wday=3, tm_yday=18, tm_isdst=0)
```
It approximately the time of the report. Perhaps the time of the request.

>>> time.gmtime(-11644246396.230042)
time.struct_time(tm_year=1601, tm_mon=1, tm_mday=3, tm_hour=15, tm_min=6, tm_sec=43, tm_wday=2, tm_yday=3, tm_isdst=0)

It is suspiciously close to the start of Win32 epoch https://devblogs.microsoft.com/oldnewthing/20090306-00/?p=18913.

@vstinner, were there some changes in time conversion that can be related to this? Or maybe Python 3.11 and 3.12 executables have different capabilities, so Win32 API returns different result? cc @zooba

**Comment by vstinner** [comment]
On Windows, `os.stat()` calls `win32_xstat_slow_impl()` which calls:

```c
    DWORD access = FILE_READ_ATTRIBUTES;
    DWORD flags = FILE_FLAG_BACKUP_SEMANTICS; /* Allow opening directories. */
    if (!traverse) {
        flags |= FILE_FLAG_OPEN_REPARSE_POINT; 
    }                   
                        
    hFile = CreateFileW(path, access, 0, NULL, OPEN_EXISTING, flags, NULL);
```

`CreateFileW()` returns `INVALID_HANDLE_VALUE` when trying to open `C:\System Volume Information` path. In this case, `basicInfo` is left unmodified, whereas the structure is allocated on the stack.

In short, `os.stat()` returns random values to timestamps (such as `st_mtime`) on `C:\System Volume Information`: when `CreateFileW()` returns `INVALID_HANDLE_VALUE`. The value depends on the state of the stack memory, in short it's "undefined" or "random".

In fact, `CreateFileW()` fails with ERROR_ACCESS_DENIED (error code 5). In this case, `attributes_from_dir()` is called and it works. Problem: `basicInfo` is not filled by this function (and is left unchanged).

cc @zooba @eryksun

**Comment by vstinner** [comment]
Note: `os.path.getmtime('C:\\System Volume Information')` returns the expected modified time if the code is run as an administrator. The problem only occurs on "access denied". For example, if I run the script in a SSH shell, I'm an administrator, and the code works as expected.

**Comment by zooba** [comment]
This is fixed in 3.13 already. See #111877.
